src/common/color/colors.ts

```diff
@@ -61,3 +61,12 @@ export const COLORS = [
 export function getColorByIndex(index: number) {
   return COLORS[index % COLORS.length];
 }
+
+export function getGraphColorByIndex(
+  index: number,
+  style: CSSStyleDeclaration
+) {
+  return (
+    style.getPropertyValue(`--graph-color-${index}`) || getColorByIndex(index)
```

Comment:
Should we use `index + 1` here for the variable name? As normal people start with 1 instead of 0?
```suggestion
    style.getPropertyValue(`--graph-color-${index}`) || getColorByIndex(index)
```

Reply:
I had thought about that as well, and I think you could argue both directions: 1..n is more logical for most non-tec users, whereas 0..n matches what we have in the coding (and we would mention that wherever we document it).

=> I am fine either way. Any other opinions here?

Reply:
tbh, changing to what 'normal people' do would be a breaking change ;-)

seriously though, I feel 0..n should be used, just like everywhere else in HA and beyond

Reply:
How would it be breaking? We are adding something that did not exist before.

Reply:
it was a joke... 
yet, we use 0...n in all user available code, be it jinja, JS or Python. Moving away from that would be new indeed.(aka breaking)

Reply:
Anybody "normal" overriding these colors can understand index starts at 0, no ? (Especially if it's in the documentation, as @spacegaier said)
Both paradigms are conceivable, but I don't see the need to add an offset...

Reply:
> 

I don't think that this is a valid comparison. Programming and low-level scripting languages of course use 0..n and we have no option to change that behavior, but here we are talking about end-user facing things that we control, so we decided to go for 1..n. Should be less confusing to the end-user.

Reply:
tbh, imho it will on the contrary  _introduce_ confusion.
Any user setting these custom colors will be used to the 0..n range in the first place. This would make it the only place in the full HA settings, backend and frontend, where 1..n is used.

of course, it's your call, and I thankfully accept the new options, just fear for a frontend issue afterwards ;-)

Reply:
Where in the frontend do we already use 0..n in end-user facing stuff? Do we already have CSS vars with 0..n that I am not aware of?

Reply:
probably not in core Ha, but all custom cards that support either JS or Jinja. And backend Jinja ofc. any mapper uses  0...n

Reply:
That is programming, this is not programming.